csvcubed/csvcubed/readers/configdeserialiser.py

```diff
@@ -0,0 +1,594 @@
+#             return None
+#
+#         else:
+#             raise Exception(f"Unexpected codelist value '{maybe_code_list}'")
+#
+#     return NewQbCodeListInCsvW(
+#         json_parent_dir
+#         / "codelists"
+#         / f"{uri_safe(column_label)}.csv-metadata.json"
+#     )
+
+
+def initialise_from_config(self, config_json: dict) -> bool:
+    """
+    Initialises a Qube from a qube json file.
+
+    param: config_json: A Qube created according to a Qube Config schema
+    """
+
+
+def _override_catalog_metadata_state(
+    catalog_metadata_json_file: Path, cube: QbCube
+) -> None:
+    with open(catalog_metadata_json_file, "r") as f:
+        catalog_metadata_dict: dict = json.load(f)
+    overriding_catalog_metadata = CatalogMetadata.from_dict(catalog_metadata_dict)
+    cube.metadata.override_with(
+        overriding_catalog_metadata,
+        overriding_keys={k for k in catalog_metadata_dict.keys()},
+    )
+
+
+def _from_config_json_dict(d: Dict,
+                           data: pd.DataFrame,
+                           json_parent_dir: Path
+                          ) -> QbCube:
+
+    metadata: CatalogMetadata = _metadata_from_dict(d)
+    # columns: List[CsvColumn] = _columns_from_config_json(d.get("columns", []), data, json_parent_dir)
+    columns: List[QbColumn] = []
+    for column_title in d.get('columns', []):
+        column_config = d['columns'].get(column_title)
+        columns.append(map_column_to_qb_component(column_title, column_config,
+                                                  data[column_title], json_parent_dir))
+
+    return Cube(metadata, data, columns)
+
+
+def _override_config_for_cube_id(config: dict, cube_id: str) -> Optional[dict]:
+    """
+    Apply cube config overrides contained inside the `cubes` dictionary to get the config for the given `cube_id`
+    """
+    # Need to do a deep-clone of the config to avoid side-effecs
+    config = copy.deepcopy(config)
+
+    config_json_id = config.get("id")
+    if config_json_id is not None and config_json_id == cube_id:
+        if "cubes" in config:
+            del config["cubes"]
+
+        return config
+    elif "cubes" in config and cube_id in config["cubes"]:
+        overrides = config["cubes"][cube_id]
+        for k, v in overrides.items():
+            config[k] = v
+
+        return config
+    else:
+        return None
+
+
+def _metadata_from_dict(config: dict) -> "CatalogMetadata":
+    creator = get_with_func_or_none(config, "creator", lambda c: str(GOV[uri_safe(c)]))
+    publisher = get_with_func_or_none(config, "publisher", lambda p: str(GOV[uri_safe(p)]))
+    themes = config.get('themes', "")
+    if isinstance(themes, str) and themes:
+        themes = [themes]
+    keywords = config.get('keywords', "")
+    if isinstance(keywords, str) and keywords:
+        keywords = [keywords]
+
+    return CatalogMetadata(
+        identifier=config.get("$id"),
```

Comment:
Do we really want to use the `$id` property? Surely `id` is good enough?

Reply:
$id seemed appropriate based on https://json-schema.org/draft/2020-12/json-schema-core.html#id-keyword
however I've updated to plain old id.